### PR TITLE
[WIP] Docker task executor implementation to worker namespace task script executor

### PR DESCRIPTION
**Pull request by paulofelipefeitosa**
**TODO**: 

1. get exit codes from the _.ec_ file.
2. recover logs output.

### CODE REVIEW

File: src/main/java/org/fogbowcloud/arrebol/execution/TaskExecutionResult.java [hunk 1]
```diff
@@ -1,47 +1,39 @@
...
 
     public RESULT getResult() {
         return this.taskResult;
     }
 
     /**
-     * It returns the exit codes for the execution of the commands.
-     * When the @{link TaskExecutor} was not able to execute the command, the exit code is
-     * is {@link TaskExecutionResult#UNDETERMINED_RESULT}.
+     * It returns the exit codes for the execution of the commands. When the @{link TaskExecutor}
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/arrebol/execution/TaskExecutionResult.java:31]: update the javadoc. It returns the Commads not the exit codes

File: src/main/java/org/fogbowcloud/arrebol/execution/creator/DockerWorkerCreator.java [hunk 2]
```diff
@@ -8,14 +9,27 @@
...
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.Collection;
 import java.util.LinkedList;
 import java.util.UUID;
 
 public class DockerWorkerCreator implements WorkerCreator{
 
     private final Logger LOGGER = Logger.getLogger(ArrebolController.class);
+    private static final String TS_NAME = "in-namespace-task-script-executor.sh";
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/arrebol/execution/creator/DockerWorkerCreator.java]: rename TS_NAME to TASK_SCRIPT_NAME there is no need to save chars here

Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/arrebol/execution/creator/DockerWorkerCreator.java]: run a code formatter in this class

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java [hunk 3]
```diff
@@ -27,78 +25,122 @@
 
     private WorkerDockerRequestHelper workerDockerRequestHelper;
     private final Logger LOGGER = Logger.getLogger(DockerTaskExecutor.class);
+    private final String tsContent;
+    private static final String WORKER_TS_FILEPATH = "/tmp/task-script-executor.sh";
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: WORKER_TASK_SCRIPT_FILEPATH

Comment (thiagomanel) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: add file separators

File: src/main/resources/in-namespace-task-script-executor.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
```
Comment (thiagomanel) [inline src/main/resources/in-namespace-task-script-executor.sh]: we need some comments in this file. bash code is not straightforward

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java [hunk 5]
```diff
@@ -27,78 +25,122 @@
...
             return taskExecutionResult;
         }
     }
+    
+    private void runScriptExecutor(String taskId, String tsFilepath) throws Exception {
+        asyncExecuteCommand("/bin/bash " + WORKER_TS_FILEPATH + " -d -tsf=" + tsFilepath, taskId);
+    }
+    
+    private void sendTaskScriptExecutor(String taskId) throws Exception {
+        LOGGER.debug("Sending Task Script Executor to Docker Worker");
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: log the task id (and the container related info too)

**Review by thiagomanel — CHANGES_REQUESTED** [review]
this review is also WIP. I'll fetch the code and (likely) send most issues latter

**Review by wesleymonte — APPROVED** [review]

**Review by wesleymonte — APPROVED** [review]

**Pull request merged**